app/src/main/java/org/mozilla/tv/firefox/telemetry/TelemetrySettingsProvider.kt

```diff
@@ -29,6 +31,8 @@ class TelemetrySettingsProvider(private val appContext: Context) : SettingsMeasu
                 appContext.serviceLocator.pinnedTileRepo.bundledTilesSize
         APP_ID -> BuildConfig.APPLICATION_ID
         PREF_REMOTE_CONTROL_NAME -> appContext.getSharedPreferences(SHARED_PREFS_KEY, 0).getStringSet(KEY_REMOTE_CONTROL_NAME, null)
+        FXA_REAUTHENTICATION_REQUIRED -> appContext.serviceLocator.fxaRepo.accountState.blockingFirst() is
```

Comment:
I verified this method gets called on the main thread and we're using a blocking call here. Since this state is handed to us by a dependency, we can't guarantee we'll always have this state when we decide to block for it, especially in future revisions of the dependency.

Can we do something less dangerous here? Can we block, with a timeout, or even assume that the state will be here and maybe a `firstOrNull` type call?

Reply:
Talked about this in person, which led to a discussion of potential edge cases here.  Adjusting the implementation accordingly.